### PR TITLE
Fix deprecated NetworkInfo class.

### LINKED ISSUE #847: NetworkInfo class deprecated
Update code to use new callback model.  https://developer.android.com/reference/android/net/NetworkInfo

### PR DESCRIPTION
**Pull request by gitaumoses4**
Fixes #847 

Updates code to use new callback model. [NetworkInfo](https://developer.android.com/reference/android/net/NetworkInfo)

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionService.java [hunk 1]
```diff
@@ -60,13 +110,17 @@ public DefaultConnectionService(Context ctx) {
...
-        return isConnectionAvailable;
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            ConnectivityManager connectivityManager = (ConnectivityManager) mConnectionContext
+                    .getSystemService(Context.CONNECTIVITY_SERVICE);
+
+            NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();
+
+            DefaultConnectionService.connectionAvailable = activeNetwork != null && activeNetwork.isConnectedOrConnecting();
+        }
+        Telemetry.emit((BaseEvent) new BaseEvent().put(TelemetryEventStrings.Key.NETWORK_CONNECTION, String.valueOf(DefaultConnectionService.connectionAvailable)));
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionService.java:121]: nit: this line is very long.  Can we factor some things to make it less so?  Is the case necessary here?

File: common/src/main/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionService.java [hunk 2]
```diff
@@ -52,6 +60,48 @@
      */
     public DefaultConnectionService(Context ctx) {
         mConnectionContext = ctx;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            this.registerNetworkCallback();
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionService.java:64]:
```suggestion
            registerNetworkCallback();
```

(We generally aren't using `this` in our code unless some variable shadowing is happening)

**Review by AdamBJohnsonx — APPROVED** [review]

File: changelog.txt [hunk 3]
```diff
@@ -6,6 +6,7 @@ V.Next
 - [PATCH] Trim() Cache Lookup Input Parameters (#1228)
 - [MINOR] Adds workaround for Mockito.openMocks() desugaring issue (#1229)
 - [PATCH] Fix bug where duplicate BrokerApplicationMetadata entries could be created for a single app (#1232)
+- Fixes deprecated NetworkInfo class (#847)
```
Comment (iambmelt) [inline changelog.txt:9]:
```suggestion
- [PATCH] Fixes deprecated NetworkInfo class (#847)
```

**Review by PaulMuriithi — APPROVED** [review]
LGTM

**Pull request merged**